gammapy/stats/fit_statistics_cython.pyx

```diff
@@ -22,13 +21,21 @@ def cash_sum_cython(np.ndarray[np.float_t, ndim=1] counts,
         Predicted counts array.
     """
     cdef np.float_t sum = 0
+    cdef np.float_t npr, lognpr
     cdef unsigned int i, ni
     ni = counts.shape[0]
     for i in range(ni):
-        if npred[i] > 0:
-            sum += npred[i]
-            if counts[i] > 0:
-                sum -= counts[i] * log(npred[i])
+        npr = npred[i]
+        if npr > 0:
+            lognpr = log(npr)
+        else:
+            npr = 1e-25
+            lognpr = -57.56
```

Comment:
It should be :
```
if npred < truncation_value:
    npr  = truncation_value
lognpr = log()
```
no ?

 Wouldn't be simpler to just avoid fittting negative norm/amplitude by default and do it only for the relevant cases ?